Cleaning chart on willDestroyElement may be required

I'm making some statistics in ember with ember-cli-nvd3 and some other charts and there is a specific bug that shows up when I resize the browser and go back and forth between routes. Canvas based Chart.js trowed an error then, and nvd3 also:

```
14d3.js:663 Error: <g> attribute transform: Expected number, "translate(NaN,5)".
```

For charts.js solution was to remove the chart on willDestroyElement (a.k.a. normal cleanup). For nvd3 similar thing happened for Angular2: http://stackoverflow.com/questions/36977176/browser-errors-on-size-change-when-ng2-nvd3-chart-is-out-of-view

--- a/addon/components/nvd3-chart.js
+++ b/addon/components/nvd3-chart.js
@@ -67,10 +67,20 @@ export default Ember.Component.extend({
       run(() => this.get('afterSetup').call(context, svgContainer, chart));
 
       // Handle window resize
-      nv.utils.windowResize(chart.update);
+      this.set('_windowResizeHandler', nv.utils.windowResize(chart.update));
 
       return chart;
     });
+  },
+
+  willDestroyElement() {
+    this._super(...arguments);
+
+    const resizeHandler = this.get('_windowResizeHandler');
+
+    if(resizeHandler && resizeHandler.clear) {
+      resizeHandler.clear();
+    }
   },
 
   evaluateOptions(chart) {

--- a/blueprints/ember-cli-nvd3/index.js
+++ b/blueprints/ember-cli-nvd3/index.js
@@ -1,4 +1,4 @@
-var RSVP = require('rsvp');
+/*jshint node:true*/
 
 module.exports = {
   description: 'Imports NVD3 dependency from Bower',
@@ -6,6 +6,6 @@ module.exports = {
   normalizeEntityName: function() {},
 
   beforeInstall: function() {
-    return this.addAddonToProject('ember-nvd3-shim@~0.1.1');
+    return this.addAddonToProject('ember-nvd3-shim@~0.1.4');
   }
 };

--- a/package.json
+++ b/package.json
@@ -39,7 +39,7 @@
     "ember-disable-prototype-extensions": "^1.1.0",
     "ember-export-application-global": "^1.0.5",
     "ember-load-initializers": "^0.5.1",
-    "ember-nvd3-shim": "~0.1.1",
+    "ember-nvd3-shim": "~0.1.4",
     "ember-resolver": "^2.0.3",
     "ember-try": "^0.2.2",
     "loader.js": "^4.0.1"
@@ -49,8 +49,7 @@
     "nvd3"
   ],
   "dependencies": {
-    "ember-cli-babel": "^5.1.6",
-    "rsvp": "^3.0.20"
+    "ember-cli-babel": "^5.1.6"
   },
   "ember-addon": {
     "configPath": "tests/dummy/config"

--- a/tests/dummy/app/controllers/index.js
+++ b/tests/dummy/app/controllers/index.js
@@ -80,11 +80,11 @@ export default Ember.Controller.extend({
 
   actions: {
     toggleChart() {
-        this.toggleProperty('isLineChart');
-      },
+      this.toggleProperty('isLineChart');
+    },
 
-      refreshData() {
-        this.set('datumBar', this.getBarData());
-      }
+    refreshData() {
+      this.set('datumBar', this.getBarData());
+    }
   }
 });

